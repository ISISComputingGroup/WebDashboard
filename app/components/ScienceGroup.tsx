import InstrumentWallCard from "@/app/components/InstrumentWallCard";
import { instList, instListEntryWithRunstatePVandValue } from "@/app/types";

export default function ScienceGroup({
  name,
  instruments,
}: {
  name: string;
  instruments: instList;
}) {
  return (
    <div
      key={name}
      className={"flex flex-col justify-center items-start w-full"}
    >
      <h2 className="w-full text-left text-gray-600 dark:text-gray-200 font-semibold text-md mt-2 py-2 ">
        {name}
      </h2>
      <div className={"flex flex-wrap gap-1"}>
        {instruments
          .sort((a, b) => a.name.localeCompare(b.name))
          .map((instrument: instListEntryWithRunstatePVandValue) => {
            return (
              <InstrumentWallCard
                key={instrument.name}
                instrument={instrument}
              />
            );
          })}
      </div>
    </div>
  );
}
