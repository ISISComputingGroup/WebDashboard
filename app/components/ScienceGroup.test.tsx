import { render } from "@testing-library/react";
import ScienceGroup from "@/app/components/ScienceGroup";

it("renders sciencegroup unchanged", () => {
  const { container } = render(
    <ScienceGroup
      name={"EXCITATIONS"}
      instruments={[
        { name: "Instrument", runstate: "RUNNING" },
        {
          name: "Instrument2",
          runstate: "WAITING",
        },
      ]}
    />,
  );
  expect(container).toMatchSnapshot();
});
