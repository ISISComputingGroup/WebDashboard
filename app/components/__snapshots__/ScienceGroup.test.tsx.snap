// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders sciencegroup unchanged 1`] = `
<div>
  <div
    class="flex flex-col justify-center items-start w-full"
  >
    <h2
      class="w-full text-left text-gray-600 dark:text-gray-200 font-semibold text-md mt-2 py-2 "
    >
      EXCITATIONS
    </h2>
    <div
      class="flex flex-wrap gap-1"
    >
      <div
        class="flex"
      >
        <a
          class="flex items-center justify-center text-center p-3 min-w-28 w-auto max-h-12 rounded-lg shadow-sm border-2 border-gray-700 dark:border-gray-200 hover:shadow-lg hover:border-black dark:hover:border-gray-700 transition-all duration-200
      bg-[#90EE90] text-black"
          href="/instrument?name=Instrument"
          target="_blank"
        >
          <div
            class="flex flex-col"
          >
            <span
              class="text-sm font-bold line-clamp-1 w-full"
            >
              Instrument
            </span>
            <span
              class="text-xs "
            >
              RUNNING
            </span>
          </div>
        </a>
      </div>
      <div
        class="flex"
      >
        <a
          class="flex items-center justify-center text-center p-3 min-w-28 w-auto max-h-12 rounded-lg shadow-sm border-2 border-gray-700 dark:border-gray-200 hover:shadow-lg hover:border-black dark:hover:border-gray-700 transition-all duration-200
      bg-[#DAA520] text-black"
          href="/instrument?name=Instrument2"
          target="_blank"
        >
          <div
            class="flex flex-col"
          >
            <span
              class="text-sm font-bold line-clamp-1 w-full"
            >
              Instrument2
            </span>
            <span
              class="text-xs "
            >
              WAITING
            </span>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
